src/js/components/SelectMultiple/stories/Variant.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+
+import { Box, SelectMultiple, Text } from 'grommet';
+
+const dummyOptions = [
+  'Apple',
+  'Orange',
+  'Banana',
+  'Grape',
+  'Melon',
+  'Strawberry',
+  'Kiwi',
+  'Mango',
+  'Raspberry',
+  'Rhubarb',
+];
+
+export const Variant = () => {
+  const [options, setOptions] = useState(dummyOptions.sort());
+  const [valueMultiple, setValueMultiple] = useState([]);
+  const [open, setOpen] = useState(false);
+
+  return (
+    // Uncomment <Grommet> lines when using outside of storybook
+    // <Grommet theme={...}>
+    <Box fill align="center" pad="large" gap="large">
+      <Text>SelectMultiple Variant</Text>
+      <SelectMultiple
+        showSelectedInline
+        placeholder="Select"
+        options={options}
+        onSearch={(text) => {
+          // The line below escapes regular expression special characters:
+          // [ \ ^ $ . | ? * + ( )
+          const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');
+
+          // Create the regular expression with modified value which
+          // handles escaping special characters. Without escaping special
+          // characters, errors will appear in the console
+          const exp = new RegExp(escapedText, 'i');
+
+          let next = [...valueMultiple];
+          next = next.filter((o) => exp.test(o));
+          let sortedOptions = dummyOptions.filter((i) => !next.includes(i));
+
+          sortedOptions = sortedOptions.filter((o) => exp.test(o));
+
+          const sortedAllOptions = next.concat(sortedOptions);
+          setOptions(sortedAllOptions);
+        }}
+        value={valueMultiple}
+        open={open}
+        onOpen={() => {
+          // setValueMultiple(intermediateValue);
+          setOpen(true);
+          let next = [...valueMultiple];
+          next.sort();
+          const sortedOptions = dummyOptions.filter((i) => !next.includes(i));
+          next = next.filter((i) => dummyOptions.includes(i));
+          sortedOptions.sort();
+          const sortedAllOptions = next.concat(sortedOptions);
+          setOptions(sortedAllOptions);
+        }}
+        onClose={() => {
```

Comment:
I think I imagined that more of this sorting would be handled within the component itself. This feels like a lot of overhead to get the desired behavior consistently across teams.

I guess this could be handled as a separate enhancement, but wondering if there could be a prop like `sort` that is a boolean that would handle the onOpen/onClose sorting happening here.

Reply:
Yeah I originally envisioned this being handled within the component but when I was implementing it I was feeling like it was making too many assumptions.
It assumes the data should be sorted alphabetically.
Any code that is based on the index of the option will not work correctly. (For example the `disabled` prop allows users to specify which options are disabled based on the index)
Custom children are passed the index of the option so there may be code in there that relies on the index of the option.

It would be nice to have a more convenient way to get the desired behavior, maybe we could have the `sort` prop but just warn that any behavior that utilizes the `index` of the option should not be used when `sort=true`?

Reply:
I would think the caller would control the sort order of the `options` up front and the SelectMultiple component would honor the relative ordering of those items when placing them in the selected portion.

Reply:
Yeah that makes sense. I added a `sort` prop in the latest commit. It sorts selected items at the top while respecting the original order of the `options` prop.